airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/http_status_error_handler.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import logging
+from typing import Any, Mapping, Optional, Tuple, Union
+
+import requests
+from airbyte_cdk.models import FailureType
+
+from .error_handler import ErrorHandler
+from .response_action import ResponseAction
+
+
+class HttpStatusErrorHandler(ErrorHandler):
+
+    _DEFAULT_ERROR_MAPPING: Mapping[Union[int, str, Exception], Mapping[str, Any]] = {
+        400: {
+            "action": ResponseAction.FAIL,
+            "failure_type": FailureType.config_error,
+            "error_message": "Bad request. Please check your request parameters.",
+        },
+        401: {
+            "action": ResponseAction.FAIL,
+            "failure_type": FailureType.config_error,
+            "error_message": "Unauthorized. Please ensure you are authenticated correctly.",
+        },
+        403: {
+            "action": ResponseAction.FAIL,
+            "failure_type": FailureType.config_error,
+            "error_message": "Forbidden. You don't have permission to access this resource.",
+        },
+        404: {
+            "action": ResponseAction.FAIL,
+            "failure_type": FailureType.system_error,
+            "error_message": "Not found. The requested resource was not found on the server.",
+        },
+        408: {
```

Comment:
I'm surprised to see that we don't retry on a transient error. In general, I think the idea of retrying and failure_type should be tightly correlated i.e. if we retry, this is a transient issue but I don't know exactly how to express that here

Reply:
I also think we should retry on transient errors since they're expected to be transient 😅 .

> if we retry, this is a transient issue but I don't know exactly how to express that here

that's an interesting point. Maybe we don't need the action at all and the failure type is enough:
- config or system error -> do not retry
- transient error -> retry according to the backoff strategy

Reply:
@girarda That was my initial thought, too, but I felt there were likely scenarios where we would decouple that relationship -- e.g. the substream 404 example.

Reply:
I think this case can be covered by mapping it to a "ignore" enum value instead of a transient error